types/convert_test.go

```diff
@@ -606,18 +606,18 @@ func (s *testTypeConvertSuite) TestConvert(c *C) {
 
 	// integer from string
 	signedAccept(c, mysql.TypeLong, "	  234  ", "234")
-	signedAccept(c, mysql.TypeLong, " 2.35e3  ", "2350")
-	signedAccept(c, mysql.TypeLong, " 2.e3  ", "2000")
-	signedAccept(c, mysql.TypeLong, " -2.e3  ", "-2000")
-	signedAccept(c, mysql.TypeLong, " 2e2  ", "200")
-	signedAccept(c, mysql.TypeLong, " 0.002e3  ", "2")
-	signedAccept(c, mysql.TypeLong, " .002e3  ", "2")
-	signedAccept(c, mysql.TypeLong, " 20e-2  ", "0")
-	signedAccept(c, mysql.TypeLong, " -20e-2  ", "0")
-	signedAccept(c, mysql.TypeLong, " +2.51 ", "3")
-	signedAccept(c, mysql.TypeLong, " -9999.5 ", "-10000")
+	signedAccept(c, mysql.TypeLong, " 2.35e3  ", "2")
```

Comment:
the correct result should be 2350?

Reply:
yes, here 2350 is the correct result. I tested some cast scenario in MySQL and found in select context `2.35e3` will be cast to 2, but if we insert `2.35e3` into an integer filed, it will be cast to 2350